src/WebJobs.Extensions/Extensions/Files/Config/FilesJobHostConfigurationExtensions.cs

```diff
@@ -57,16 +59,45 @@ public FilesExtensionConfig(FilesConfiguration filesConfig)
                 _filesConfig = filesConfig;
             }
 
+            private FileInfo GetFileInfo(FileAttribute attribute)
+            {
+                string boundFileName = attribute.Path;
+                string filePath = Path.Combine(_filesConfig.RootPath, boundFileName);
+                FileInfo fileInfo = new FileInfo(filePath);
+                return fileInfo;
+            }
+
+            public FileStream GetFileStream(FileAttribute attribute)
+            {
+                var fileInfo = GetFileInfo(attribute);
+                if ((attribute.Access == FileAccess.Read) && !fileInfo.Exists)
+                {
+                    return null;
+                }
+
+                return fileInfo.Open(attribute.Mode, attribute.Access);
+            }
+
+            public Stream Convert(FileAttribute attribute)
+            {
+                return GetFileStream(attribute);
+            }
+
             public void Initialize(ExtensionConfigContext context)
             {
                 if (context == null)
                 {
                     throw new ArgumentNullException("context");
                 }
 
+                var rule = context.AddBindingRule<FileAttribute>();
+                rule.BindToInput<FileInfo>(this.GetFileInfo);
+                rule.BindToInput<FileStream>(this.GetFileStream);
+                rule.BindToStream(this, FileAccess.ReadWrite);
+
                 context.Config.RegisterBindingExtensions(
-                    new FileTriggerAttributeBindingProvider(_filesConfig, context.Trace),
-                    new FileAttributeBindingProvider(_filesConfig, context.Config.NameResolver));
+                    new FileTriggerAttributeBindingProvider(_filesConfig, context.Trace)
```

Comment:
Does this binding provider now handle trigger as well as non-trigger bindings (e.g. the input/output file bindings). If so, then it might make sense to remove "trigger" from its name.

Reply:
No change to  triggers yet.  So we still need the TriggerBindingProviders. I had a separate idea that may improve this, but will tackle later.